src/components/Form/form.tsx

```
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import usersFromServer from '../../api/users';
+import { Todo, User } from '../Types/types';
+import { getUserById } from '../../App';
+
+interface Props {
+  onAdd: (event: Todo) => void;
+  todos: Todo[];
+}
+
+interface FormValues {
+  title: string;
+  selected: number;
+}
+
+export const Form: React.FC<Props> = ({ onAdd, todos: currentTodos }) => {
+  const [selected, setSelected] = useState(0);
+  const [title, setTitle] = useState('');
+  const [selectedError, setSelectedError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const formValues: FormValues = {
+    title: title,
+    selected: selected,
+  };
+
+  function handleTitleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    if (titleError) {
+      setTitleError(false);
+    }
+
+    setTitle(event.target.value);
+  }
+
+  function handleSelectedChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    if (selectedError) {
+      setSelectedError(false);
+    }
+
+    setSelected(+event.target.value);
+  }
+
+  function cleanForm() {
+    setTitle('');
+    setSelected(0);
+    setTitleError(false);
+    setSelectedError(false);
+  }
+
+  function validateData(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (!formValues.title.trim()) {
+      setTitleError(true);
+    }
+
+    if (formValues.selected === 0) {
+      setSelectedError(true);
+    }
+
+    if (!formValues.title.trim() || formValues.selected === 0) {
+      return;
+    }
+
+    const newIndex: number = Math.max(...currentTodos.map(todo => todo.id)) + 1;
+
+    onAdd({
+      id: newIndex,
+      title: formValues.title,
+      userId: formValues.selected,
+      completed: false,
+      user: getUserById(formValues.selected) as User,
```

Using a type assertion `as User` is risky here. The `getUserById` function can return `null`, and this assertion would hide a potential error from TypeScript, which could lead to a runtime crash. It's safer to handle the possibility of a `null` user, even if other validation is in place.